src/db/result_set.cr

```diff
@@ -99,6 +104,16 @@ module DB
       {% end %}
     end
 
+    private def internal_read(**types : **T) forall T
+      {% begin %}
+        NamedTuple.new(
+          {% for name, type in T %}
+            {{ name }}: read({{type.instance}}),
```

Comment:
I kind of like that the naming is defined by the order not the names. Sometimes your DB will not use userscore_case identifiers and you want to convert them.